Library/Homebrew/language/node.rb

```diff
@@ -0,0 +1,32 @@
+module Language
+  module Node
+    def self.npm_cache_config
+      "cache=#{HOMEBREW_CACHE}/npm_cache\n"
+    end
+
+    def self.setup_npm_environment
+      # explicitly use our npm and node-gyp executables instead of the user
+      # managed ones in HOMEBREW_PREFIX/lib/node_modules which might be broken
+      ENV.prepend_path "PATH", "#{Formula["node"].opt_libexec}/npm/bin"
+      # explicitly set npm's cache path to HOMEBREW_CACHE/npm_cache to fix
+      # issues caused by overriding $HOME (long build times, high disk usage)
+      # https://github.com/Homebrew/brew/pull/37#issuecomment-208840366
+      npmrc = Pathname.new("#{ENV["HOME"]}/.npmrc")
+      npmrc.write npm_cache_config unless npmrc.exist?
+    end
+
+    def self.std_npm_install_args(libexec)
+      setup_npm_environment
+      # tell npm to not install .brew_home by adding it to .npmignore
+      open('.npmignore', 'a') { |f| f.write( "\n.brew_home\n") }
```

Comment:
Are we certain that we're in the right directory when opening `.npmignore` with a relative path? Also, if we can be sure this file already exists, using [`Pathname#append_lines`](https://github.com/Homebrew/brew/blob/da34fba151ee33c1a2e14ab21ee0dc4ea451cc0f/Library/Homebrew/extend/pathname.rb#L145-L149) might be a neater way to do what is being done here, e.g. `Pathname.new(".npmignore").append_lines(".brew_home")`.

Just some thoughts, feel free to ignore if not applicable. (I've only skimmed the discussion in this long-ish PR.)